### PR TITLE
Shop: getItems() now returns item ID, add method for buying item with item name

### PR DESCRIPTION
**Pull request by illumineawake**
Shop: getItems() now returns item id instead of widget id
add methods for purchasing from Shop based on item name

WorldArea: add toString() method

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/coords/WorldArea.java [hunk 1]
```diff
@@ -772,4 +772,16 @@ public WorldPoint getRandom()
 		return new WorldPoint(ThreadLocalRandom.current().nextInt(x, x + width),
 				ThreadLocalRandom.current().nextInt(y, y + height), plane);
 	}
+
+	@Override
+	public String toString()
```
Comment (buracc) [inline runelite-api/src/main/java/net/runelite/api/coords/WorldArea.java:777]: whats this for?

Comment (illumineawake) [reply runelite-api/src/main/java/net/runelite/api/coords/WorldArea.java:777]: Just to try match consistency with WorldPoint.toString() and help with debugging. But mainly for my own benefit so happy to remove it if you need?

**Comment by d-o-g** [comment]
Why does `getItems()` return ids in the first place? Instead of a `List<Item>`

File: runelite-api/src/main/java/net/runelite/api/coords/WorldArea.java (hunk 1 above)
Comment (buracc) [reply runelite-api/src/main/java/net/runelite/api/coords/WorldArea.java:777]: I'd like to keep modifications to the RL api as low as possible, so if this is purely for debugging then please remove it. If you'd like to debug, simply use the getters, I dont think this class needs a toString

**Pull request merged**